packages/contracts/contracts/Router.sol

```diff
@@ -0,0 +1,220 @@
+// SPDX-License-Identifier: UNLICENSED
```

Comment:
need to improve comments and code coverage here and in the factory before merging:

```
  Router.sol              |        0 |        0 |        0 |        0 |... 203,207,218 |
  RouterFactory.sol       |     6.25 |        0 |    16.67 |     6.25 |... 63,70,71,75 |
  ```

Reply:
Router.sol              |    73.33 |    52.94 |    66.67 |    73.91 |... 104,107,148 |
  RouterFactory.sol       |      100 |     87.5 |      100 |      100 |                |
  
  the coverage beforeEach was broken for both contracts. Here is the updated coverage.